pkg/storegateway/series_chunks.go

```diff
@@ -365,24 +387,49 @@ func (c *loadingSeriesChunksSetIterator) Next() (retHasNext bool) {
 
 			err := c.chunkReaders.addLoad(chunk.blockID, chunk.ref, i, j)
 			if err != nil {
-				c.err = errors.Wrap(err, "preloading chunks")
-				return false
+				return nil, false, errors.Wrap(err, "preloading chunks")
 			}
 		}
 	}
 
 	// Create a batched memory pool that can be released all at once.
 	chunksPool := pool.NewSafeSlabPool[byte](chunkBytesSlicePool, chunkBytesSlabSize)
 
-	err := c.chunkReaders.load(nextSet.series, chunksPool, c.stats)
+	err = c.chunkReaders.load(nextSet.series, chunksPool, c.stats)
 	if err != nil {
-		c.err = errors.Wrap(err, "loading chunks")
-		return false
+		return nil, false, errors.Wrap(err, "loading chunks")
 	}
 
 	nextSet.chunksReleaser = chunksPool
-	c.current = nextSet
-	return true
+
+	if c.ignoreNativeHistograms {
+		seriesWriteIdx := 0
```

Comment:
can please you move this to its own function?

Reply:
Yep!